src/services/firebase/firestore.ts

```diff
@@ -677,4 +699,132 @@ export class FirestoreServices {
 
     return fileURLs;
   };
+
+  private deleteUnreadAndTypingUser = async (conversationId: string) => {
+    if (!this.userId || !conversationId) {
+      console.error('User ID or conversation ID is missing');
+      return;
+    }
+
+    const conversationRef = firestore()
+      .collection<ConversationData>(
+        this.getUrlWithPrefix(`${FireStoreCollection.conversations}`)
+      )
+      .doc(conversationId);
+
+    try {
+      const conversationDoc = await conversationRef.get();
+      if (conversationDoc.exists) {
+        const conversationData = conversationDoc.data();
+        if (
+          conversationData &&
+          conversationData.unRead &&
+          conversationData.unRead[this.userId]
```

Comment:
I think only need check **conversationData?.unRead?.[this.userId]**

Reply:
Yes, I updated